src/main/profiling_info.cpp

```diff
@@ -9,69 +9,79 @@ using namespace duckdb_yyjson; // NOLINT
 
 namespace duckdb {
 
+ProfilingInfo::ProfilingInfo(const profiler_settings_t &n_settings, const idx_t depth) : settings(n_settings) {
+	// Expand.
+	if (depth == 0) {
+		settings.insert(MetricsType::QUERY_NAME);
+	} else {
+		settings.insert(MetricsType::OPERATOR_TYPE);
+	}
+	for (const auto &metric : settings) {
+		Expand(expanded_settings, metric);
+	}
+
+	// Reduce.
+	if (depth == 0) {
+		auto op_metrics = DefaultOperatorSettings();
+		for (const auto metric : op_metrics) {
+			settings.erase(metric);
+		}
+	} else {
+		auto root_metrics = DefaultRootSettings();
+		for (const auto metric : root_metrics) {
+			settings.erase(metric);
+		}
+	}
+	ResetMetrics();
+}
+
 profiler_settings_t ProfilingInfo::DefaultSettings() {
 	return {MetricsType::QUERY_NAME,           MetricsType::BLOCKED_THREAD_TIME,     MetricsType::CPU_TIME,
```

Comment:
Unrelated to this PR - but can we perhaps turn `profiler_settings_t` into a bitset instead of a set? That should be faster.